### DISCUSSION TITLE
Jupiter's Moons

### DISCUSSION THREAD

**Discussion by meshula** [cosinekitty/astronomy#97] [General]
On Jean Meeus' algorithm.

The given calculations give the positions of the four great satellites of Jupiter with respect to the planet as seen from the Earth. The apparent rectangular coordinates X and Y of the satellites will be measured from the center of Jupiter's disc, in units of the planet's equatorial radius. X is measured positively to the west of Jupiter. Y is positive to the north as indicated by the northern rotational pole. The low accuracy method is sufficient for identifying the moons when viewed from a telescope, or making a "wavy line" diagram. The high accuracy method should be used when calculating eclipses, transits, and so on.

The date and instant are converted to the Julian Day. Because Jupiter's rotation axis almost exactly perpendicular to the planet's orbital plane around the Sun, it is not necessary to correct for the Sun's aberration in the calculation of the axis. The specific calculations (corresponding to Chapter 43 of Meeus' book, may be found here: https://github.com/soniakeys/meeus/blob/master/v3/jupiter/jupiter.go. In those calculations, DE and DS refer to the planetocentric declinations of the earth and sun. P is the position angle of Jupiter's northern rotation pole.

The high accuracy calculations compensate for relativistic effects as a ray of light passes near Jupiter's gravity well!

LMK what might be useful to expand upon.

**Comment by meshula** [comment]
The Position Angle P of a body (a1, d1) with respect to another body (a2, d2), can be found from

```
tan P = sin (a1 - a2)
             ---------------------
              cosd2 tand1 - sind2 cos(a1 - a2)
```
If the denominator of the fraction is negative then P lies in the range 90-270 degrees

P is positive counter clockwise:
```

             North
              |
              |
        1 *   |
           \ P|
            \ |
             \|
              * 2
              |

```

**Comment by cosinekitty** [comment]
Thank you @meshula ! I think this is enough information to get started. I also found the PDF of the Meeus book, which I am studying. Ordinarily I would pay for my reference sources, but because the book is out of print and the publisher went out of business, and the asking price for used copies is beyond my reasonable budget, I don't see how this use harms the author or the (defunct) publisher. I will gladly send Jean Meeus some money if I can figure out how to contact him! At any rate, I will start working on calculating Jupiter's moons in Astronomy Engine this week, in a separate branch. I will post here to keep anyone updated who might be interested.

**Comment by meshula** [comment]
With his retirement some years ago, and all of his books out of print, we are in a really strange place in computational astronomy, cobbling together knowledge from rare sources, and no one place to go to. We need a new book and reference, that teaches how all the pieces fit together...

**Comment by cosinekitty** [comment]
I have started development in a new branch, for anyone who wants to follow the process:
https://github.com/cosinekitty/astronomy/tree/jupiter_moons
So far I have just defined Jupiter radius constants. I intend to pick up speed over the next week or so.

**Reply by cosinekitty** [reply]
I finally took a look at Stellarium's source code. It calculates position and velocity vectors for Jupiter's moons based on some FORTRAN code ported to C from:

https://ftp.imcce.fr/pub/ephem/satel/galilean/L1/L1.2/

This appears to be a more recent analytic model. Perhaps it is more accurate than the Meeus one? I'm considering implementing both and seeing if I can get them to match. That will be a good exercise that I'm doing coordinate transforms and unit conversion correctly.
